### ISSUE TITLE
ttnn.reshape 2D reshaping in LeNet Data Parallel results low PCC

### ISSUE THREAD

**Issue by sabira-mcw** [tenstorrent/tt-metal#15422]
**Describe the bug**
Reshaping a tensor to 2D in a data parallel setup with ttnn.reshape results in a low PCC value of -0.0015

**To Reproduce**
1. pick the test from [Commit](https://github.com/tenstorrent/tt-metal/compare/main...sabira/test_reshape_mesh)
2. Run  ```pytest tests/ttnn/unit_tests/operations/test_reshape_mesh.py```

**Expected behavior**
The PCC expected to be at a normal level during reshaping the tesnor.

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Devices : WH

**Comment by sabira-mcw** [comment]
Tried the following workaround in the LeNet model. 
```
x = ttnn.reshape(x, (x.shape[0], -1))
x = ttnn.from_device(x)
x = ttnn.to_device(x, device=mesh_device)
x = ttnn.to_layout(x, layout=ttnn.TILE_LAYOUT) 
```
It strangely results in a significant boost to the  PCC 0.99 in the model.

**Comment by saichandax** [comment]
@ntarafdar , Can we assign this to someone, as discussed about getting a fix for this issue by this week?

cc: @boris-drazic

**Comment by jvegaTT** [comment]
I think there is a problem with the mesh tensor and either how it splits or collects the tensors across the devices. The above splits the tensor such that each device gets a (16,16,5,5) tensor to be reshaped to (16,400) before being re-merged. I have verified that in a single device tensor the above reshape works well in tile on my branch ( https://github.com/tenstorrent/tt-metal/pull/15572) but it still fails with multi-device tensors

**Comment by ntarafdar** [comment]
Thanks for checking @jvegaTT , @cfjchu can you look at this?

**Comment by cfjchu** [comment]
Taking a look

**Comment by zzigler-tt** [comment]
@cfjchu Hello, is there any update on this item, I believe it is blocking progress on a couple models that the development team is working on. If we can review and provide a timeline for remediation it would be great.

**Comment by cfjchu** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/122298491?s=64&amp;v=4">@cfjchu](https://github.com/cfjchu) Hello, is there any update on this item, I believe it is blocking progress on a couple models that the development team is working on. If we can review and provide a timeline for remediation it would be great.

what's the priority of this issue? I have a few other things on my plate right now.

**Comment by prajaramanTT** [comment]
@cfjchu @sabira-mcw is this still an open issue ? If not, can you please close this ? Thanks

**Comment by Sudharsan-V** [comment]
@cfjchu, @prajaramanTT, This issue still exists.

**Comment by kkeerthana0573** [comment]
@cfjchu, 
This issue is blocks three of the models. 


cc: @zzigler-tt

**Comment by zzigler-tt** [comment]
@prajaramanTT Can you work with the team to understand the priority and the LoE for remediating this issue? Thanks

**Comment by cfjchu** [comment]
I'm marking this as a P1. No immediate plans to fix this on our end and it seems you guys have a work-around: https://github.com/tenstorrent/tt-metal/issues/15422#issuecomment-2506433361

**Comment by sabira-mcw** [comment]
@cfjchu Understood that there are no immediate plans to fix this, but resolving it is critical for the trace+2CQS implementation, as the host-device transfer will be blocked. Prioritizing this would be helpful to move forward.

cc: @zzigler-tt

**Comment by saichandax** [comment]
@cfjchu ,
Can you give a timeline for this issue, so that we can plan Data parallel and trace+2cqs implementations accordingly for Mnist, ConvMnist and LeNet models?

cc: @zzigler-tt , @mbahnasTT

**Comment by zzigler-tt** [comment]
@cfjchu I understand this is a lower priority, although it would be great to get a sense of the remediation timeline, and/or if there's other folks who may be able to assist. I'd like to unblock these models so we can close out open items. Thanks for your help. 

@adifrancescoTT FYI

**Comment by saichandax** [comment]
We have a fix for the issue in #19688.
Need to test and confirm.

@MohammedTaherMcW, Did you verify if the fix resolved the UTs in this issue description?

**Comment by MohammedTaherMcW** [comment]
> We have a fix for the issue in [#19688](https://github.com/tenstorrent/tt-metal/pull/19688). Need to test and confirm.
> 
> [@MohammedTaherMcW](https://github.com/MohammedTaherMcW), Did you verify if the fix resolved the UTs in this issue description?

Yes, I tested above example with the PR (#19688) changes and it's working.

**Comment by saichandax** [comment]
Ok. We'll test in the models. Thank you.

@sabira-mcw , @kkeerthana0573 , @Sudharsan-V - Please note to test our models, when we re-visit them, and then we can close this issue.

**Comment by ntarafdar** [comment]
@saichandax can we close this issue?
